src/layers/ServerCluster.js

```diff
@@ -227,10 +237,15 @@ class ServerCluster extends Cluster {
         )
     }
 
-    getVisibleTiles = () =>
-        this.getSourceCacheTiles()
+    getVisibleTiles = async () => {
+        while (!this.isTilesUpdated()) {
+            await new Promise(r => setTimeout(r, 100))
```

Comment:
Why `100`? Was this amount chosen for a particular reason?

Reply:
Also, would it be possible to wait for the promise to resolve instead of polling every X ms?

Reply:
100 was just a number I thought was reasonable :-)

The promise will resolve when the tiles are updated, but I haven't found a way to do it without polling. I would expect Mapbox GL JS to fire the "moveend" event _after_ the  map is updated, but this is not the case.

Reply:
Thanks for the answer, makes sense. Disregard my comment 👍